packages/server/customer-os-webhooks/service/utils.go

```diff
@@ -24,3 +31,243 @@ func CallEventsPlatformGRPCWithRetry[T any](fn func() (T, error)) (T, error) {
 	}
 	return response, err
 }
+
+type EmailChannelData struct {
+	Subject   string   `json:"Subject"`
+	InReplyTo []string `json:"InReplyTo"`
+	Reference []string `json:"Reference"`
+}
+
+func buildEmailChannelData(subject string, references, inReplyTo []string) (*string, error) {
+	emailContent := EmailChannelData{
+		Subject:   subject,
+		InReplyTo: utils.EnsureEmailRfcIds(inReplyTo),
+		Reference: utils.EnsureEmailRfcIds(references),
+	}
+	jsonContent, err := json.Marshal(emailContent)
+	if err != nil {
+		return nil, err
+	}
+	jsonContentString := string(jsonContent)
+
+	return &jsonContentString, nil
+}
+
+func GetWhitelistedDomain(domain string, whitelistedDomains []commonEntity.WhitelistDomain) *commonEntity.WhitelistDomain {
+	for _, allowedOrganization := range whitelistedDomains {
+		if strings.Contains(domain, allowedOrganization.Domain) {
+			return &allowedOrganization
+		}
+	}
+	return nil
+}
+
+func (s *syncEmailService) ConvertToUTC(datetimeStr string) (time.Time, error) {
```

Comment:
this should be in common module, we have already a method to parse string to date, can be enhanced with all below layouts